engines/base/src/main/java/org/alfresco/transform/base/TransformManager.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * #%L
+ * Alfresco Transform Core
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * -
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ * -
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * -
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * -
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.transform.base;
+
+import org.alfresco.transform.common.TransformException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+/**
+ * Allows {@link CustomTransformer} implementations to interact with the base t-engine.
+ */
+public interface TransformManager
+{
+    /**
+     * @return the id of the request.
+     */
+    String getRequestId();
+
+    /**
+     * Allows a {@link CustomTransformer} to use a local source {@code File} rather than the supplied {@code InputStream}.
+     * The file will be deleted once the request is completed. To avoid creating extra files, if a File has already
+     * been created by the base t-engine, it is returned.
+     * If possible this method should be avoided as it is better not to leave content on disk.
+     * @throws IllegalStateException if this method has already been called.
+     */
+    File createSourceFile();
+
+    /**
+     * Allows a {@link CustomTransformer} to use a local target {@code File} rather than the supplied {@code OutputStream}.
+     * The file will be deleted once the request is completed. To avoid creating extra files, if a File has already
+     * been created by the base t-engine, it is returned.
+     * If possible this method should be avoided as it is better not to leave content on disk.
+     * @throws IllegalStateException if this method has already been called. A call to {@link #respondWithFragment(Integer, boolean)}
+     *         allows the method to be called again.
+     */
+    File createTargetFile();
```

Comment:
Missing `throws` in method declaration or should not be added to javadocs.

Reply:
done